documentation/style-guide.rst

```diff
@@ -196,6 +199,35 @@ language):
     achieve the same effect.  This assures that files are flushed and file
     descriptor resources are released in a timely manner.
 
+
+Author attribution
+==================
+
+For new documentation, do not use a byline (naming the author of the document).
+Explicit attribution tends to discourage other users from updating community
+documentation.
```

Comment:
I'd be interested to see background on this, if possible. I've personally never felt deterred by a byline, and it is a nice way to credit someone who's written a major guide or document. However, perhaps there is a more quantitative study that we could reference if introducing this guidance?

Reply:
I do feel uneasy about changing docs that are attributed to a specific author. It feels like I'm putting words in their mouth that they may not agree with.

Reply:
I think it depends on the type of document and the type of change.  For example, updating module docs is easier than updating a howto, since in the howto the original author might have had a more specific vision and goals.

On the other hand, having a byline is also useful to get in touch with the original author and ask for advice and reviews before making major changes (this could technically be replaced by `git blame` and `CODEOWNERS`).

Reply:
> it is a nice way to credit someone who's written a major guide or document

How do we credit someone who has written a major code changes? I don't think we add their name to the top of module name, so why would this be any different? This is an open source documentation not a blog.

> having a byline is also useful to get in touch with the original author and ask for advice and reviews before making major changes (this could technically be replaced by git blame and CODEOWNERS)

As you said, this can be done by `git blame` and `codeowners`. However I would like for us to rethink the notion that the original author is the only one capable of reviewing/giving advice. Their advice may be sought, but should not be the only advice that matters, and should not be the only advice we rely on. Other contributors' review and advice should be taken into consideration equally.

Reply:
My experience in CPython is that personal ownership of areas has sometimes given rise to conflicts when other contributors wanted to meaningfully participate (without always bowing to the original author's preferences). So I'm in favor of this guidline.

Reply:
I agree with @pitrou.

Reply:
> I do feel uneasy about changing docs that are attributed to a specific author

@JelleZijlstra I am supportive of this for existing documentation. However, this PR is concerning new documentation :smile:

Reply:
> I'd be interested to see background on this, if possible. I've personally never felt deterred by a byline, and it is a nice way to credit someone who's written a major guide or document. However, perhaps there is a more quantitative study that we could reference if introducing this guidance?

@AA-Turner The byline style recommendation is related to new documentation. We have seen that bylines in certain areas of the documentation has caused conflicts where an individual acts as owner of the community resource. We see similar behavior when we look at academic publishing: https://pubmed.ncbi.nlm.nih.gov/31161378/ 

My general feeling is that reference documentation should not have bylines. As others have mentioned `git blame` or `git log` can offer recognition. If someone creates a new "How to", I think it would be reasonable to recognize in release notes/what's new as we do code contributions.

Reply:
> I am supportive of this for existing documentation. However, this PR is concerning new documentation

True. We could also consider removing bylines from existing documentation, or replacing them with something more indefinite like "Originally written by Jelle Zijlstra; maintained by the community". However, that's out of scope for this change.

Reply:
I like the suggestion @JelleZijlstra for future wording: 
>  "Originally written by Jelle Zijlstra; maintained by the community"